test/integration/node-specific/client_close.test.ts

```diff
@@ -91,34 +131,195 @@ describe.skip('MongoClient.close() Integration', () => {
+                  };
+                  await runScriptAndGetProcessInfo('timer-monitor-interval', config, run);
+                }
+              );
             });
 
             describe('after a heartbeat fails', () => {
-              it.skip('the new monitor interval timer is cleaned up by client.close()', async () => {});
+              it(
+                'the new monitor interval timer is cleaned up by client.close()',
+                metadata,
+                async () => {
+                  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
+                  const run = async function ({ MongoClient, uri, expect, sleep, getTimerCount }) {
+                    const heartbeatFrequencyMS = 2000;
+                    const client = new MongoClient('mongodb://fakeUri', { heartbeatFrequencyMS });
+                    let heartbeatHappened = false;
+                    client.on('serverHeartbeatFailed', () => (heartbeatHappened = true));
+                    client.connect();
+                    await sleep(heartbeatFrequencyMS * 2.5);
+                    expect(heartbeatHappened).to.be.true;
+
+                    function getMonitorTimer(servers) {
+                      for (const server of servers) {
+                        return server[1]?.monitor.monitorId.timerId;
+                      }
+                    }
+                    const servers = client.topology.s.servers;
+                    expect(getMonitorTimer(servers)).to.exist;
+                    await client.close();
+                    expect(getMonitorTimer(servers)).to.not.exist;
+
+                    expect(getTimerCount()).to.equal(0);
+                  };
+                  await runScriptAndGetProcessInfo('timer-heartbeat-failed-monitor', config, run);
+                }
+              );
             });
           });
         });
 
         describe('Connection Monitoring', () => {
           describe('Node.js resource: Socket', () => {
-            it.skip('no sockets remain after client.close()', metadata, async function () {});
+            it('no sockets remain after client.close()', metadata, async function () {
+              const run = async function ({ MongoClient, uri, expect }) {
+                const client = new MongoClient(uri);
+                await client.connect();
+
+                // returns all active tcp endpoints
+                const connectionMonitoringReport = () =>
+                  process.report
+                    .getReport()
+                    .libuv.filter(r => r.type === 'tcp')
+                    .map(r => r.remoteEndpoint);
+
+                const servers = client.topology?.s.servers;
+                // assert socket creation
+                for (const server of servers) {
+                  const { host, port } = server[1].s.description.hostAddress;
+                  expect(connectionMonitoringReport()).to.deep.include({ host, port });
+                }
```

Comment:
```suggestion
                for (const [, server] of servers) {
                  const { host, port } = server.s.description.hostAddress;
                  expect(connectionMonitoringReport()).to.deep.include({ host, port });
                }
```

Just so that the variable "server" is actually the server and not the tuple that we have to access the second item in.